File: src/main/java/info/tritusk/inductivemachine/InductiveMachine.java
package info.tritusk.inductivemachine;

import ic2.api.event.TeBlockFinalCallEvent;
import ic2.core.block.TeBlockRegistry;
import info.tritusk.inductivemachine.block.BlockInductional;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

@Mod(modid = "inductivemachine", name = "Inductive Machine", version = "0.0.1", useMetadata = true, dependencies = "required-after:IC2@[2.6.76,)")
public class InductiveMachine {
	
	static {
		MinecraftForge.EVENT_BUS.register(InductiveMachine.class);
	}
	
	@Mod.Instance("inductivemachine")
	public static InductiveMachine instance;
	
	@SubscribeEvent
	public static void onIC2TeRegFinalCall(TeBlockFinalCallEvent event) {
		TeBlockRegistry.addAll(BlockInductional.class, BlockInductional.ID);
	}
	
	@Mod.EventHandler
	public void postInit(FMLPostInitializationEvent event) {
		TeBlockRegistry.get(BlockInductional.ID).setCreativeTab(new CreativeTabs("inductive_machine") {
			@Override
			public Item getTabIconItem() {
				return Item.getItemFromBlock(TeBlockRegistry.get(BlockInductional.ID));
			}		
		});
	}

}

File: src/main/kotlin/info/tritusk/inductivemachine/InductiveMachine.kt
package info.tritusk.inductivemachine

import ic2.core.block.TeBlockRegistry
import net.minecraft.util.ResourceLocation
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
import ic2.api.event.TeBlockFinalCallEvent as TeBlockReg

@Mod(modid = "inductivemachine", useMetadata = true, modLanguage = "kotlin", modLanguageAdapter = "net.shadowfacts.forgelin.KotlinAdapter")
object InductiveMachine {

    @JvmField val BLOCK_ID: ResourceLocation = ResourceLocation("inductivemachine", "machine")
    @JvmField val GUI: ResourceLocation = ResourceLocation("inductivemachine", "gui/inductive_machine.xml")

    init {
        MinecraftForge.EVENT_BUS.register(object : Any() {
            @SubscribeEvent fun onIC2TeReg(event: TeBlockReg) = TeBlockRegistry.addAll(BlockInductional::class.java, BLOCK_ID)
        })
    }

}
